pkg/kubectl/cmd/apply.go

```diff
@@ -345,7 +345,11 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, opti
 
 			if string(patchBytes) == "{}" {
 				count++
-				cmdutil.PrintSuccess(shortOutput, out, info.Object, false, "unchanged")
+
+				// Do not print unchanged message if an output format other than "name" is specified
+				if len(output) == 0 || shortOutput {
```

Comment:
This is the opposite of the if condition below `if len(output) > 0 && !shortOutput` so worth making it a var.

Reply:
will do